backends/dpdk/run-dpdk-ptf-test.py

```diff
@@ -0,0 +1,324 @@
+            # Terminate the switch process and emit its output in case of failure.
+            testutils.kill_proc_group(self.switch_proc)
+
+    
+    def create_TAPs(self,  insecure_mode: bool = True) -> int:
+        """ Create TAPs with gNMI """
+        testutils.log.info(
+            "---------------------- Creating TAPs ----------------------",
+        )
+        for index in range(self.options.num_taps):
+            tap_name = f"TAP{index}"
+            cmd = (f"{self.options.ipdk_recipe}/install/bin/gnmi-ctl set "
+                   f"device:virtual-device,name:{tap_name},pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:1500,port-type:TAP "
+                   f"-grpc_use_insecure_mode={insecure_mode}")
+            _, returncode = testutils.exec_process(cmd, timeout=5)
+            if returncode != testutils.SUCCESS:
+                testutils.log.error("Failed to create TAP")
+                return returncode
+            _, returncode = testutils.exec_process(f"ifconfig {tap_name} up")
+            if returncode != testutils.SUCCESS:
+                testutils.log.error("Failed to activate TAP interface")
+                return returncode
+        return returncode
+            
+
+    def compile_program(self,info_name:Path, bf_rt_schema:Path,context:Path, dpdk_spec: Path ) -> int:
+
+        """ Create /pipe directory """
+        _, returncode = testutils.exec_process(f"mkdir {self.options.testdir.joinpath('pipe')}", timeout=30)
+        if returncode != testutils.SUCCESS:
+            testutils.log.error("Failed to create /pipe directory")
+            return returncode
+
+
+        """Compile the input P4 program using p4c-dpdk."""
+        testutils.log.info("---------------------- Compile with p4c-dpdk ----------------------")
+        compilation_cmd = (
+            #f"{self.options.p4c_dir}/build/p4c-dpdk --arch pna --target dpdk \
+            f"p4c-dpdk --arch pna --target dpdk \
+            --p4runtime-files {info_name} \
+            --bf-rt-schema {bf_rt_schema} \
+            --context {context} \
+            --tdi-builder-conf {self.options.testdir.joinpath(Path(self.options.p4_file.name).with_suffix('.conf'))}\
+            -o {dpdk_spec} {self.options.p4_file}"
+        )
+        _, returncode = testutils.exec_process(compilation_cmd, timeout=30)
+        if returncode != testutils.SUCCESS:
+            testutils.log.error("Failed to compile the P4 program %s.", self.options.p4_file)
+        return returncode
+
+    def run_infrap4d(self,insecure_mode: bool = True) -> testutils.subprocess.Popen:
+        """Start infrap4d and return the process handle."""
+        testutils.log.info(
+            "---------------------- Start infrap4d ----------------------",
+        )
+        run_infrap4d_cmd = (
+            f"{self.options.ipdk_recipe}/install/sbin/infrap4d "
+            f"-grpc_open_insecure_mode={insecure_mode}"
+        )
+        self.switch_proc = testutils.open_process(run_infrap4d_cmd)
```

Comment:
You should be able to pass environment variables from the arguments like this 
https://stackoverflow.com/a/4453495/3215972

I would be surprised if that did not work.